tests/components/hue_ble/test_config_flow.py

```diff
@@ -0,0 +1,627 @@
+"""Test the Hue BLE config flow."""
+
+from unittest.mock import AsyncMock, PropertyMock, patch
+
+from homeassistant import config_entries
+from homeassistant.components.hue_ble.config_flow import (
+    CannotConnect,
+    InvalidAuth,
+    NotFound,
+    ScannerNotAvailable,
+)
+from homeassistant.components.hue_ble.const import DOMAIN
+from homeassistant.const import CONF_MAC, CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+from homeassistant.helpers import device_registry as dr
+
+from . import HUE_BLE_SERVICE_INFO, TEST_DEVICE_MAC, TEST_DEVICE_NAME
+
+from tests.components.bluetooth import generate_ble_device
+
+
+async def test_form(hass: HomeAssistant, mock_setup_entry: AsyncMock) -> None:
+    """Test form with good data."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {}
+
+    with (
+        patch(
+            "homeassistant.components.hue_ble.config_flow.async_ble_device_from_address",
+            return_value=generate_ble_device(TEST_DEVICE_NAME, TEST_DEVICE_MAC),
+        ),
+        patch(
+            "homeassistant.components.hue_ble.config_flow.async_scanner_count",
+            return_value=1,
+        ),
+        patch(
+            "homeassistant.components.hue_ble.config_flow.HueBleLight.connect",
+            return_value=True,
+        ),
+        patch(
+            "homeassistant.components.hue_ble.config_flow.HueBleLight.authenticated",
+            return_value=True,
+            new_callable=PropertyMock,
+        ),
+        patch(
+            "homeassistant.components.hue_ble.config_flow.HueBleLight.connected",
+            return_value=True,
+            new_callable=PropertyMock,
+        ),
+        patch(
+            "homeassistant.components.hue_ble.config_flow.HueBleLight.poll_state",
+            return_value=(True, []),
+        ),
+    ):
+        result = await hass.config_entries.flow.async_configure(
```

Comment:
Move this to a fixture in conftest.py, and use it for all tests to avoid repetition